lib/workers/repository/process/write.ts

```diff
@@ -40,7 +47,30 @@ export async function writeUpdates(
     }
     addMeta(meta);
     const branchExisted = branchExists(branch.branchName);
-    const res = await processBranch(branch);
+    let branchCache = {} as BranchCache;
+    if (branchExisted) {
+      branchCache =
+        cachedBranches?.find((br) => br.branchName === branch.branchName) ??
+        ({} as BranchCache);
+
+      if (
+        config.repositoryCache === 'enabled' &&
+        Object.keys(branchCache).length === 0
+      ) {
+        logger.debug(`No branch cache found for ${branch.branchName}`);
+      }
+    }
```

Comment:
```suggestion
    if (branchExisted && config.repositoryCache === 'enabled') {
      branchCache =
        cachedBranches?.find((br) => br.branchName === branch.branchName) ??
        ({} as BranchCache);

      if (Object.keys(branchCache).length === 0) {
        logger.debug(`No branch cache found for ${branch.branchName}`);
      }
    }
```

Does this make more sense?